apps/api/src/domain/domainModelValidators/contextValidators/validateEdgeConnection/validateIdentityEdgeConnection.ts

```diff
@@ -0,0 +1,95 @@
+import { isMemberContextTheIdentityContext } from '.';
+import { InternalError } from '../../../../lib/errors/InternalError';
+import {
+    EdgeConnection,
+    EdgeConnectionMember,
+    EdgeConnectionMemberRole,
+} from '../../../models/context/edge-connection.entity';
+import { ResourceType } from '../../../types/ResourceType';
+import IncompatibleIdentityConnectionMembersError from '../../errors/context/IncompatibleIdentityConnectionMembersError';
+import LonelyIdentityContextInEdgeconnectionError from '../../errors/context/LonelyIdentityContextInEdgeConnectionError';
+import SelfConnectionCannotUseIdentityContextError from '../../errors/context/SelfConnectionCannotUseIdentityContextError';
+
+/***
+ * This assumes that you have 2 members from a dual Edge Connection and you want
+ * to find either the to or the from member.
+ */
+export const findMemberWithRole = (
```

Comment:
Refactor so that the logic is
- validateSelfEdgeConnection
- validateDualEdgeConnection